### PR TITLE
desc: make clippy happy

### PR DESCRIPTION
**Pull request by overvenus**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -11,6 +11,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#![cfg_attr(test, deny(warnings))]
```
Comment (siddontang) [inline src/lib.rs]: why denying warning?

Comment (overvenus) [inline src/lib.rs]: So that something like `make clippy happy` can be prevent due to ci failure.

Comment (siddontang) [inline src/lib.rs]: @BusyJay 
Is it ok?

Comment (BusyJay) [inline src/lib.rs]: It's ok. But I prefer put this to configuration file like `.travis.yml`. I usually fix the warnings at last (after all the tests passed).

Comment (overvenus) [inline src/lib.rs]: I think it would be confusing to contributors, why local tests passed but ci failed?

Comment (BusyJay) [inline src/lib.rs]: I don't think CI is guaranteed to run the same test cases with the same configuration in same environment as contributors'. It's normal that local tests succeed but CI fails. What is actually needed is clear fail message and logs.

**Comment by siddontang** [comment]
we may assign special rust and clippy version to let travis pass now.

File: src/lib.rs (hunk 1 above)
Comment (siddontang) [inline src/lib.rs]: I prefer in `.travis.yml` too.
Most of users use stable rust version, so they may still meet "local tests succeed but CI fails".

**Comment by siddontang** [comment]
PTAL @BusyJay

**Comment by overvenus** [comment]
Please wait a  moment, it's still in progress ...

**Comment by overvenus** [comment]
PTAL @siddontang @BusyJay

File: .travis.yml [hunk 2]
```diff
@@ -3,7 +3,11 @@ language: rust
...
-  - nightly
+
+matrix:
+  include:
+  - rust: nightly
+    env: FEATURE="--features dev"
 
 script:
-  - cargo test
+  - cargo test $FEATURE 2>&1 | awk 'BEGIN {flag = 0} {print} /warning/ {flag = 1}  END{if (flag>0) {print "Please fix warnings"; exit 1}}'
```
Comment (BusyJay) [inline .travis.yml]: Please refer https://github.com/pingcap/tikv/blob/master/travis-build/test.sh#L30

**Comment by siddontang** [comment]
LGTM

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**